data-pipeline/druid-events-validator/src/main/java/org/ekstep/ep/samza/service/TelemetryValidatorService.java

```diff
@@ -71,4 +66,29 @@ private String getInvalidFieldName(String errorInfo) {
         String[] pointer = fields[3].split("\"pointer\":");
         return pointer[1].substring(0, pointer[1].length() - 1);
     }
+
+    private String getSchema(Event event) {
+        String telemetrySchemaFilePath;
+        String summaryEventSchemaFilepath;
+        StringBuilder sb = new StringBuilder();
+        telemetrySchemaFilePath = MessageFormat.format("{0}/{1}/{2}", TELEMETRY_EVENT_SCHEMA_BASE_PATH, event.version(), event.schemaName());
+        summaryEventSchemaFilepath = MessageFormat.format("{0}/{1}/{2}", SUMMARY_EVENT_SCHEMA_BASE_PATH, event.version(), event.schemaName());
+        InputStream is = this.getClass().getResourceAsStream(event.isSummaryEvent() ? summaryEventSchemaFilepath : telemetrySchemaFilePath);
+        BufferedReader br;
+        if (is == null) {
+            telemetrySchemaFilePath = MessageFormat.format("{0}/{1}/{2}", TELEMETRY_EVENT_SCHEMA_BASE_PATH, event.version(), DEFAULT_EVENT_SCHEMA_FILE);
+            summaryEventSchemaFilepath = MessageFormat.format("{0}/{1}/{2}", SUMMARY_EVENT_SCHEMA_BASE_PATH, event.version(), DEFAULT_EVENT_SCHEMA_FILE);
+            is = this.getClass().getResourceAsStream(event.isSummaryEvent() ? summaryEventSchemaFilepath : telemetrySchemaFilePath);
+        }
+        br = new BufferedReader(new InputStreamReader(is));
+        String line;
+        try {
```

Comment:
How are we making sure that the resource is closed successfully and there are no memory leaks?